pkg/columns/formatter/json/json.go

```diff
@@ -157,18 +156,18 @@ func (f *Formatter[T]) getPrinter(level int, cols []*column[T]) func(entry *T, b
 	// Better name?
 	handlePrefix := func() {
 		if !first {
-			funcs = append(funcs, func(entry *T, buf *encodeState) {
+			funcs = append(funcs, func(indent string, entry *T, buf *encodeState) {
 				buf.WriteByte(',')
 				if f.options.prettyPrint {
-					buf.WriteString("\n" + indent)
+					buf.WriteString("\n" + indent + "  ")
```

Comment:
`indent` is not always `"  "`: it's empty when printing the root element, and can be longer when printing a child element. The `"  "` there are used to indent the single field in the object, while indent is used to for the whole object. 

Just to run a quick experiment, I changed it to used `"--"` instead of  `"  "` in that line and in L170. this is the result:

```
[
  {
  --"name": {
    --"first": "Alice",
    --"second": "Taylor"
    },
  --"age": 32,
  --"size": 1.74,
  --"balance": 1000,
  --"canDance": true
  }
]
```

Reply:
Everything is indented using two spaces. That's not configurable when using `-o jsonpretty`